driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java

```diff
@@ -968,17 +958,15 @@ private void tryLock(final Timeout timeout) throws MongoTimeoutException {
         }
 
         /**
-         * Returns {@code timeoutNanos} if {@code timeoutNanos} is negative, otherwise returns 0 or a positive value.
-         *
-         * @param timeoutNanos Use a negative value for an infinite timeout,
-         *                     in which case {@link Condition#awaitNanos(long)} is called with {@link Long#MAX_VALUE}.
+         * @param timeoutNanos See {@link Timeout#startNow(long)}.
+         * @return The remaining duration as per {@link Timeout#remainingOrInfinite(TimeUnit)} if waiting ended early either
+         * spuriously or because of receiving a signal.
          */
-        private long awaitNanos(final long timeoutNanos) {
+        private long awaitNanos(final Condition condition, final long timeoutNanos) throws MongoInterruptedException {
```

Comment:
This change makes the behavior of this method consistent with `Timeout` and allows to see explicitly what condition it waits for on the calling side.